src/test/java/org/alfresco/heartbeat/HBDataCollectorServiceImplTest.java

```diff
@@ -52,76 +61,168 @@
     public void setUp()
     {
         mockDataSenderService = mock(HBDataSenderService.class);
-        dataCollectorService = new HBDataCollectorServiceImpl(true);
+
+        dataCollectorService = spy(new HBDataCollectorServiceImpl(true));
         dataCollectorService.setHbDataSenderService(mockDataSenderService);
     }
 
-
-
     @Test
-    public void testInitialEnabledEqualsDefaultState()
+    public void testInitialEnabledEqualsDefaultState() throws Exception
     {
-        HBDataCollectorService dataCollectorService = new HBDataCollectorServiceImpl(true);
+        HBDataCollectorServiceImpl dataCollectorService = new HBDataCollectorServiceImpl(true);
         assertTrue(dataCollectorService.isEnabledByDefault());
 
         dataCollectorService = new HBDataCollectorServiceImpl(false);
         assertFalse(dataCollectorService.isEnabledByDefault());
     }
 
+    private void activateHB(HBDataCollectorServiceImpl dataCollectorService, boolean activate)
+    {
+        LicenseDescriptor mockLicenseDescriptor = mock(LicenseDescriptor.class);
+        when(mockLicenseDescriptor.isHeartBeatDisabled()).thenReturn(!activate);
+        dataCollectorService.onLicenseChange(mockLicenseDescriptor);
+    }
+
+    @Test
+    public void testWrongCronExpression() throws Exception
+    {
+        HBDataCollectorServiceImpl dataCollectorService = new HBDataCollectorServiceImpl(true);
+
+        activateHB(dataCollectorService, true);
+
+        Scheduler scheduler = spy(StdSchedulerFactory.getDefaultScheduler());
+        dataCollectorService.setScheduler(scheduler);
+
+        // try with correct cron expression
+        SimpleHBDataCollector collector1 = new SimpleHBDataCollector("collector1");
+        collector1.setCronExpression("0 0 0/1 * * ?");
+
+        dataCollectorService.registerCollector(collector1);
+
+        verifyNumbersOfUnschedulesAndSchedules(scheduler, 1, 1);
```

Comment:
Could verify that the job is scheduled with the expected job name and that it actually is in the scheduler